### ISSUE TITLE
error in 3.8: Cannot read property 'forEach' of undefined

### ISSUE THREAD

**Issue by chen23** [F5Networks/f5-appsvcs-extension#52]
##### Do you already have an issue opened with F5 support?
no existing SR

##### ISSUE TYPE

 - Bug Report

##### AS3 BUILD/ VERSION

```
{
    "version": "3.8.0",
    "release": "3",
    "schemaCurrent": "3.8.0",
    "schemaMinimum": "3.0.0"
}
```

##### BIGIP VERSION

```
Sys::Version
Main Package
  Product     BIG-IP
  Version     13.1.1.2
  Build       0.0.4
  Edition     Point Release 2
  Date        Thu Oct 11 15:32:21 PDT 2018
```

##### SUMMARY

when submitting a declaration that previously worked in 3.7 it fails in 3.8 with the error
```
Cannot read property 'forEach' of undefined
```

##### STEPS TO REPRODUCE

```
{
    "class": "AS3",
    "action": "deploy",
    "persist": true,
    "declaration": {
        "class": "ADC",
        "schemaVersion": "3.2.0",
        "id": "test",
        "AS3Demo": {
            "class": "Tenant",
            "defaultRouteDomain": 0,
            "DemoApplication": {
                "class": "Application",
                "template": "https",
                "serviceMain": {
                    "class": "Service_HTTPS",
                    "remark": "Accepts HTTPS/TLS connections on port 443",
                    "clientTLS": {
                       "bigip": "/Common/serverssl"
                     },
                    "virtualAddresses": ["10.1.10.10"],
                    "redirect80": false,
                    "pool": "custom_ssl_pool",
                    "profileTCP": {
			            "egress": "wan",
            			"ingress": { "use": "TCP_Profile" } },
                    "profileHTTP": { "use": "custom_http_profile" },
                    "serverTLS": { "bigip": "/Common/clientssl" },
                    "persistenceMethods": [],
                    "policyWAF": {
                       "bigip": "/Common/asm-policy-linux-high-security_policy"
                    },
                    "securityLogProfiles": [{ "bigip":"/Common/Log all requests"}]
                },
                "plain_HTTP": {
                    "class": "Service_HTTP",
                    "remark": "Accepts HTTP connections on port 80",
                    "virtualAddresses": ["10.1.10.10"],
                    "pool": "custom_pool",
                    "profileTCP": {
			            "egress": "wan",
            			"ingress": { "use": "TCP_Profile" } },
                    "profileHTTP": { "use": "custom_http_profile" },
                    "policyEndpoint": "forward_policy"
                },
                "custom_pool": {
                    "class": "Pool",
                    "monitors": ["http"],
                    "members": [{
                        "servicePort": 8080,
                        "serverAddresses": ["192.168.128.11","192.168.128.12","192.168.128.13"]
                    }]
                },
               "special_pool": {
                    "class": "Pool",
                    "monitors": ["http"],
                    "members": [{
                        "servicePort": 8080,
                        "serverAddresses": ["192.168.128.14"]
                    }]
                },
               "custom_ssl_pool": {
                    "class": "Pool",
                    "monitors": ["https"],
                    "members": [            {
              "serverAddresses": [
                "192.168.128.14"
              ],
              "servicePort": 8443
            },
            {
              "serverAddresses": [
                "192.168.128.15"
              ],
              "servicePort": 8443
            },
            {
              "serverAddresses": [
                "192.168.128.16"
              ],
              "servicePort": 8443
            }]
                },
                "custom_http_profile": {
                    "class": "HTTP_Profile",
                    "xForwardedFor": true
                },
                "TCP_Profile": {
        			"class": "TCP_Profile",
        			"idleTimeout": 60 },
        			       "forward_policy": {
        "class": "Endpoint_Policy",
        "rules": [{
          "name": "forward_to_pool",
          "conditions": [{
            "type": "httpUri",
            "path": {
              "operand": "starts-with",
              "values": ["/headers/"]
            }
          }],
          "actions": [{
            "type": "forward",
            "event": "request",
            "select": {
              "pool": {
                "use": "special_pool"
              }
            }
          }]
        },{
          "name": "redirect_secure",
          "conditions": [{
            "type": "httpUri",
            "path": {
              "operand": "starts-with",
              "values": ["/txt"]
            }
          }],
          "actions": [{
            "type": "httpRedirect",
            "event": "request",
            
              "location": "tcl:https://[getfield [HTTP::host] \":\" 1][HTTP::uri]"
            
          }]
        }]
      }
            }
        }
    }
}
```

##### EXPECTED RESULTS

not error
##### ACTUAL RESULTS

error
```
{
    "code": 500,
    "declarationFullId": "test",
    "message": "Cannot read property 'forEach' of undefined"
}
```

**Comment by kaustriaf5** [comment]
Hi @chen23 thanks for reporting, repro confirmed. We have opened issue 842 for this.

**Comment by kaustriaf5** [comment]
@chen23 The fix is now available starting with [3.8.1](https://github.com/F5Networks/f5-appsvcs-extension/releases/tag/v3.8.1). Closing this issue.
